### DISCUSSION TITLE
pcigale-plots fail with negative flux error for upperlimit

### DISCUSSION THREAD

**Discussion by Bovornpratch** [mboquien/cigale#25] [Q&A] [answered]
Hi, 

I doing some fitting for some AGNs which contain both no data values and upper limit values.  Following the discussion #8 I set the nodata to (nan, nan) and set upper limits to (ulim, -ulim). 

However, after fitting I try to run pcigale-sed to examin the plots but cigale returns. 

ValueError: 'yerr' must not contain negative values

One work around I have is setting upperlimits to (ulim, ulim). 

I attach an example data and config at this link 
[test.zip](https://github.com/mboquien/cigale/files/9720855/test.zip)

**Comment by mboquien** [comment]
Hello. I tried your example and I did not find any issue. However, it could be that it is a bug I already fixed. What version are you using exactly?

![Screenshot from 2022-10-06 05-58-10](https://user-images.githubusercontent.com/444815/194211157-6511dbd5-1651-4f96-af1a-43cd997c19b8.png)

**Reply by Bovornpratch** [reply]
Hi, thanks for the quick reply
I'm using this version.
CIGALE version: 2022.1 — Python version: 3.8.10 — Platform: linux-x86_64

**Reply by mboquien** [reply]
In have made quite a few corrections since then. Could you try with this development version if that works for you? https://gitlab.lam.fr/cigale/cigale/-/archive/develop/cigale-develop.tar.gz

**Reply by Bovornpratch** [reply]
Hi, No good either. 

This now sounds like a machine specific problem. I have confirmed that it is due to the negative value of the errors. I replaced the upper limits with (nan,nan) and he plots are produced. 

Can you tell me you python3 version?

**Reply by mboquien** [reply]
This is really curious. I am running Python 3.10, but I do not think it should make any difference.

**Comment by Bovornpratch** [comment, marked as answer]
Hi, After some debugging I think I figured it out.

The issue is not python but actually matplotlib 3.6.0
I found that the portion of the code that is triggering this is the part that plots the residuals.

line 442 of init.py in /plot_types/sed/__ini__t.py
`mask = np.where(obs_fluxes > 0.0) 
  ax2.errorbar(filters_wl[mask], 
                       (obs_fluxes[mask] - mod_fluxes[mask]) / obs_fluxes[mask], 
                       yerr=obs_fluxes_err[mask] /  obs_fluxes[mask], 
                       marker="_", label="(Obs-Mod)/Obs", 
                       color="k", capsize=0.0, ls="None", lw=1, )
`



In Matplotlib 3.6.0 they enforce that xerr and yerr must be >0
https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.errorbar.html
Thus this breaks the pcigale-plots since the mask still includes the upperlimit values.

Downgrading to any version <3.6.0 should solve this issue for the time being. Perhaps it may be a good idea to enforce versions in the setup?
